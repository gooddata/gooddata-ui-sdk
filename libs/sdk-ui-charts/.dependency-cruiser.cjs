const depCruiser = require("../../common/config/dep-cruiser/default.config");

options = {
    forbidden: [
        ...depCruiser.DefaultRules,
        ...depCruiser.DefaultSdkRules,
        ...depCruiser.PublicLibraryRules,
        depCruiser.isolatedSubmodule("chart-interfaces", "src/interfaces"),
        depCruiser.moduleWithDependencies("highcharts", "src/highcharts", ["src/interfaces"]),
        //depCruiser.moduleWithDependencies("charts", "src/charts", ["src/highcharts", "src/interfaces"]),
        /* These appear as desired deps for the two modules; currently there are validation errors. The
         * refactoring should by driven by need to remove these validation errors.
        depCruiser.moduleWithDependencies(
            "chart-options-builder",
            "src/highcharts/chartTypes/_chartOptions",
            [
                "src/highcharts/chartTypes/_util/*",
                "src/interfaces",
                "src/highcharts/typings/*",
                "src/highcharts/constants/*",
                "src/highcharts/lib",
            ],
        ),
        depCruiser.moduleWithDependencies("chart-creators", "src/highcharts/chartTypes/_chartCreators", [
            "src/highcharts/chartTypes/_util/*",
            "src/interfaces",
            "src/highcharts/typings/*",
            "src/highcharts/constants/*",
            "src/highcharts/lib",
        ]),*/
    ],
    options: depCruiser.DefaultOptions,
};

module.exports = options;
