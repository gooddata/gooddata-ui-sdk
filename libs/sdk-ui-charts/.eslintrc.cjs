// (C) 2020 GoodData Corporation

const { tsOverride } = require("@gooddata/eslint-config/tsOverride");

module.exports = {
    extends: ["@gooddata/eslint-config/esm-react-vitest"],
    overrides: [
        tsOverride(__dirname, {
            "@typescript-eslint/no-unsafe-assignment": "off",
            "@typescript-eslint/no-unsafe-call": "off",
            "@typescript-eslint/no-unnecessary-type-assertion": "off",
            "@typescript-eslint/no-unsafe-member-access": "off",
            "@typescript-eslint/no-unsafe-argument": "off",
            "@typescript-eslint/no-unsafe-return": "off",
            "@typescript-eslint/require-await": "off",
            "@typescript-eslint/no-redundant-type-constituents": "off",
            "@typescript-eslint/naming-convention": "off",
            "@typescript-eslint/unbound-method": "off",
            "@typescript-eslint/no-unsafe-enum-comparison": "off",
        }),
    ],
};
