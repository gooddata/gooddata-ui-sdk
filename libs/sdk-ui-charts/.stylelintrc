{
  "extends": [
    "../../.stylelintrc"
  ],
  "rules": {
    "max-nesting-depth": 4
  }
}
