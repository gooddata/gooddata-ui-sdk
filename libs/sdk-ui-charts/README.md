# GoodData.UI SDK - Charts

## License

(C) 2017-2021 GoodData Corporation

This project is under commercial license. See [LICENSE](LICENSE).
