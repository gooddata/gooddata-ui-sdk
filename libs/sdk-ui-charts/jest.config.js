// (C) 2019 GoodData Corporation
process.env = Object.assign(process.env, { NODE_ICU_DATA: "node_modules/full-icu" });
const base = require("../../common/config/jest/jest.config.base.js");
module.exports = {
    ...base,
    testEnvironment: "jsdom",
    setupFilesAfterEnv: ["<rootDir>/jest.setup.ts", "jest-enzyme"],
};
