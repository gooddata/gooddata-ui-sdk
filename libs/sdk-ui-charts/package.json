{
    "name": "@gooddata/sdk-ui-charts",
    "version": "8.0.0-alpha.14",
    "description": "GoodData.UI SDK - Charts",
    "main": "dist/index.js",
    "repository": "https://github.com/gooddata/gooddata-ui-sdk/tree/master/libs/sdk-ui-charts",
    "author": "GoodData Corporation",
    "license": "BSD-3-Clause",
    "files": [
        "dist",
        "src",
        "styles",
        "LICENSE"
    ],
    "config": {
        "tslint": "./{__mocks__,src,stories}/{,**/}*.{ts,tsx}"
    },
    "scripts": {
        "prepublishOnly": "npm run build",
        "build": "bash scripts/build.sh",
        "build-dev": "bash scripts/build.sh --dev",
        "dev": "bash scripts/build.sh --dev-watch",
        "test": "jest --watch",
        "test-once": "jest",
        "test-ci": "JEST_JUNIT_OUTPUT=./ci/results/test-results.xml jest --ci --config jest.ci.js",
        "tslint": "tslint -p . -t verbose $npm_package_config_tslint",
        "tslint-ci": "mkdir -p ./ci/results && tslint -p . -t checkstyle -o ./ci/results/tslint-results.xml $npm_package_config_tslint",
        "prettier-check": "prettier --check '{src,stories,styles,__mocks__}/**/*.{ts,tsx,json,scss,md,yaml,html}'",
        "prettier-write": "prettier --write '{src,stories,styles,__mocks__}/**/*.{ts,tsx,json,scss,md,yaml,html}'",
        "dep-cruiser": "depcruise --validate .dependency-cruiser.js --output-type err-long src/",
        "dep-cruiser-ci": "depcruise --validate .dependency-cruiser.js --output-type err-html src/ >./ci/results/dep-cruiser.html",
        "validate": "tsc -p tsconfig.build.json --noEmit && npm run tslint && npm run prettier-check",
        "validate-ci": "npm run tslint-ci && npm run prettier-check"
    },
    "typings": "dist/index.d.ts",
    "dependencies": {
        "@ag-grid-community/all-modules": "22.1.1",
        "@ag-grid-community/react": "22.1.2",
        "@formatjs/intl-pluralrules": "~1.3.7",
        "@gooddata/goodstrap": "^66.4.0",
        "@gooddata/js-utils": "^3.4.0",
        "@gooddata/numberjs": "^3.2.3",
        "@gooddata/sdk-backend-spi": "^8.0.0-alpha.14",
        "@gooddata/sdk-model": "^8.0.0-alpha.14",
        "@gooddata/sdk-ui": "^8.0.0-alpha.14",
        "classnames": "^2.2.6",
        "custom-event": "^1.0.1",
        "date-fns": "^2.8.1",
        "downshift": "^3.4.7",
        "fixed-data-table-2": "^0.7.17",
        "highcharts": "7.1.1",
        "highcharts-grouped-categories": "^1.1.3",
        "hoist-non-react-statics": "^3.3.0",
        "http-status-codes": "^1.3.0",
        "invariant": "^2.2.2",
        "lodash": "^4.17.15",
        "moment": "^2.24.0",
        "react-day-picker": "7.3.0",
        "react-dom-factories": "^1.0.2",
        "react-intl": "^3.6.0",
        "react-measure": "^2.3.0",
        "react-responsive": "^8.0.1",
        "react-window": "^1.8.5",
        "uuid": "^3.3.3"
    },
    "peerDependencies": {
        "react": "^16.10.0",
        "react-dom": "^16.10.0"
    },
    "devDependencies": {
        "@gooddata/reference-workspace": "^8.0.0-alpha.0",
        "@gooddata/sdk-backend-mockingbird": "^8.0.0-alpha.14",
        "@gooddata/tslint-config": "1.0.0",
        "@types/classnames": "2.2.3",
        "@types/enzyme": "^3.10.3",
        "@types/enzyme-adapter-react-16": "^1.0.5",
        "@types/hoist-non-react-statics": "^3.3.1",
        "@types/invariant": "^2.2.29",
        "@types/jest": "^25.1.2",
        "@types/lodash": "^4.14.123",
        "@types/node": "^12.12.26",
        "@types/raf": "^3.4.0",
        "@types/react": "^16.9.11",
        "@types/react-dom": "^16.9.3",
        "@types/react-measure": "2.0.5",
        "@types/react-portal": "4.0.1",
        "@types/react-responsive": "^8.0.2",
        "@types/react-window": "^1.8.1",
        "@types/uuid": "^3.4.5",
        "dependency-cruiser": "^6.0.0",
        "enzyme": "^3.10.0",
        "enzyme-adapter-react-16": "^1.15.1",
        "foundation-sites": "^5.5.3",
        "full-icu": "^1.3.0",
        "jest": "^25.1.0",
        "jest-enzyme": "^7.1.2",
        "jest-junit": "^3.0.0",
        "node-sass": "^4.13.0",
        "node-sass-magic-importer": "^5.3.2",
        "prettier": "~1.19.1",
        "raf": "^3.4.1",
        "ts-jest": "^25.2.0",
        "tslint": "^5.14.0",
        "tslint-config-prettier": "^1.18.0",
        "typescript": "3.7.4"
    }
}
