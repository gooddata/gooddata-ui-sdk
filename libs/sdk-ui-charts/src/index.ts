// (C) 2007-2020 GoodData Corporation
export * from "./charts";
export * from "./highcharts";
