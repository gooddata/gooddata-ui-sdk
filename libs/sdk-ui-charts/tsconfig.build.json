{
    "extends": "../tsconfig.react.json",
    "exclude": ["**/tests", "**/test"],
    "include": ["src/**/*", "src/**/*.json", "scripts/validateCss.js", "src/highcharts.d.ts"],
    "compilerOptions": {
        "baseUrl": ".",
        "rootDir": "./src",
        "outDir": "./esm"
    }
}
