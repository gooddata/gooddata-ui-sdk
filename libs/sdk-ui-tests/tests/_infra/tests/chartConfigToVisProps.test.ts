// (C) 2020 GoodData Corporation
import { chartConfigToVisProperties } from "../chartConfigToVisProps.js";
import { IChartConfig } from "@gooddata/sdk-ui-charts";
import { describe, it, expect } from "vitest";

describe("chartConfigToControls", () => {
    const Scenarios: Array<[string, IChartConfig]> = [
        ["convert xaxis alignment - low", { xaxis: { name: { position: "low" } } }],
        ["convert xaxis alignment - middle", { xaxis: { name: { position: "middle" } } }],
        ["convert xaxis alignment - high", { xaxis: { name: { position: "high" } } }],
        ["convert secondary xaxis alignment - low", { secondary_xaxis: { name: { position: "low" } } }],
        ["convert secondary xaxis alignment - middle", { secondary_xaxis: { name: { position: "middle" } } }],
        ["convert secondary xaxis alignment - high", { secondary_xaxis: { name: { position: "high" } } }],
        ["convert yaxis alignment - low", { yaxis: { name: { position: "low" } } }],
        ["convert yaxis alignment - middle", { yaxis: { name: { position: "middle" } } }],
        ["convert yaxis alignment - high", { yaxis: { name: { position: "high" } } }],
        ["convert secondary yaxis alignment - low", { secondary_yaxis: { name: { position: "low" } } }],
        ["convert secondary yaxis alignment - middle", { secondary_yaxis: { name: { position: "middle" } } }],
        ["convert secondary yaxis alignment - high", { secondary_yaxis: { name: { position: "high" } } }],
    ];

    it.each(Scenarios)("should %s", (_desc, input) => {
        expect(chartConfigToVisProperties(input)).toMatchSnapshot();
    });
});
