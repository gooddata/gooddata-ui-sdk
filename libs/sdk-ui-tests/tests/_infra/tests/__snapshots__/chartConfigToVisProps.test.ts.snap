// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`chartConfigToControls should convert secondary xaxis alignment - high 1`] = `
Object {
  "controls": Object {
    "secondary_xaxis": Object {
      "name": Object {
        "position": "right",
      },
    },
  },
}
`;

exports[`chartConfigToControls should convert secondary xaxis alignment - low 1`] = `
Object {
  "controls": Object {
    "secondary_xaxis": Object {
      "name": Object {
        "position": "left",
      },
    },
  },
}
`;

exports[`chartConfigToControls should convert secondary xaxis alignment - middle 1`] = `
Object {
  "controls": Object {
    "secondary_xaxis": Object {
      "name": Object {
        "position": "center",
      },
    },
  },
}
`;

exports[`chartConfigToControls should convert secondary yaxis alignment - high 1`] = `
Object {
  "controls": Object {
    "secondary_yaxis": Object {
      "name": Object {
        "position": "top",
      },
    },
  },
}
`;

exports[`chartConfigToControls should convert secondary yaxis alignment - low 1`] = `
Object {
  "controls": Object {
    "secondary_yaxis": Object {
      "name": Object {
        "position": "bottom",
      },
    },
  },
}
`;

exports[`chartConfigToControls should convert secondary yaxis alignment - middle 1`] = `
Object {
  "controls": Object {
    "secondary_yaxis": Object {
      "name": Object {
        "position": "center",
      },
    },
  },
}
`;

exports[`chartConfigToControls should convert xaxis alignment - high 1`] = `
Object {
  "controls": Object {
    "xaxis": Object {
      "name": Object {
        "position": "right",
      },
    },
  },
}
`;

exports[`chartConfigToControls should convert xaxis alignment - low 1`] = `
Object {
  "controls": Object {
    "xaxis": Object {
      "name": Object {
        "position": "left",
      },
    },
  },
}
`;

exports[`chartConfigToControls should convert xaxis alignment - middle 1`] = `
Object {
  "controls": Object {
    "xaxis": Object {
      "name": Object {
        "position": "center",
      },
    },
  },
}
`;

exports[`chartConfigToControls should convert yaxis alignment - high 1`] = `
Object {
  "controls": Object {
    "yaxis": Object {
      "name": Object {
        "position": "top",
      },
    },
  },
}
`;

exports[`chartConfigToControls should convert yaxis alignment - low 1`] = `
Object {
  "controls": Object {
    "yaxis": Object {
      "name": Object {
        "position": "bottom",
      },
    },
  },
}
`;

exports[`chartConfigToControls should convert yaxis alignment - middle 1`] = `
Object {
  "controls": Object {
    "yaxis": Object {
      "name": Object {
        "position": "center",
      },
    },
  },
}
`;
